packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,260 @@
+
+  /**
+   * Existing schedule group name
+   *
+   * @default 'default'
+   */
+  readonly groupName?: string;
+
+  /**
+   * The customer managed KMS key that EventBridge Scheduler will use to encrypt and decrypt payload.
+   *
+   * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/encryption-rest.html
+   *
+   * @default - use automatically generated KMS key
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * The expression that defines when the schedule runs.
+   *
+   * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/schedule-types.html
+   */
+  readonly scheduleExpression: string;
+
+  /**
+   * The timezone in which the scheduling expression is evaluated.
+   *
+   * @default - UTC
+   */
+  readonly timezone?: string;
+
+  /**
+   * The date, in UTC, after which the schedule can begin invoking its target.
+   * Depending on the schedule's recurrence expression, invocations might occur on, or after, the StartDate you specify. EventBridge Scheduler ignores StartDate for one-time schedules.
+   *
+   * @default - No start date
+   */
+  readonly startDate?: Date;
+
+  /**
+   * Specifies whether the schedule is enabled or disabled.
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The Amazon Resource Name (ARN) of the target.
+   *
+   * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-targets.html
+   */
+  readonly target: string;
+
+  /**
+   * The IAM role that EventBridge Scheduler will use for this target when the schedule is invoked.
+   */
+  readonly role: iam.IRole;
+
+  /**
+   * The input to the target.
+   *
+   * @default - EventBridge Scheduler delivers a default notification to the target
+   */
+  readonly input?: string;
+
+  /**
+   * The retry policy settings
+   *
+   * @default - Do not retry
+   */
+  readonly retryPolicy?: RetryPolicy;
+
+  /**
+   * Dead letter queue for failed events
+   *
+   * @default - No dead letter queue
+   */
+  readonly deadLetterQueue?: sqs.IQueue;
+}
+
+/**
+ * The information about the retry policy settings
+ */
+export interface RetryPolicy {
+  /**
+   * The maximum number of retry attempts to make before the request fails.
+   */
+  readonly maximumRetryAttempts: number;
+
+  /**
+   * The maximum amount of time to continue to make retry attempts.
+   */
+  readonly maximumEventAge: Duration;
+}
+
+/**
+ * Create a new AWS EventBridge Scheduler schedule
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/APIReference/API_CreateSchedule.html
+ */
+export class EventBridgeSchedulerCreateScheduleTask extends sfn.TaskStateBase {
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    super(scope, id, props);
+
+    this.validateProps(props);
+
```

Comment:
The necessary [IAM permissions](https://docs.aws.amazon.com/scheduler/latest/UserGuide/encryption-rest.html) should be applied if `kmsKey` is specified.

Reply:
Do we need encryption permissions? ([docs](https://docs.aws.amazon.com/scheduler/latest/UserGuide/encryption-rest.html#key-management-iam-permissions-execution-role)) 🤔

Reply:
That's right! I've updated to use `gramtDecrypt()` method.